vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3842,9 +3842,12 @@ public void recalculateColumnWidths() {
         // the client side grid in the beforeClientResponse hook, we need to
         // match this here so that the column width recalculation runs after the
         // data was updated.
-        getElement().getNode().runWhenAttached(ui -> ui.beforeClientResponse(
-                this,
-                ctx -> getElement().callJsFunction("recalculateColumnWidths")));
+        getElement().getNode().runWhenAttached(ui -> {
+            ui.getInternals().getStateTree().collectChanges(ignore -> {
+            });
+            ui.beforeClientResponse(this, ctx -> getElement()
+                    .callJsFunction("recalculateColumnWidths"));
+        });
```

Comment:
With this change, the column doesn't actually hide at all:

https://user-images.githubusercontent.com/1222264/187182996-bb1f1c2d-b235-46e9-9929-0f4e40015c94.mp4

Same example but with the previous implementation hides the column properly:

https://user-images.githubusercontent.com/1222264/187183077-fac81945-6637-409f-ae32-3800560e9ba9.mp4

Please add a test that checks that the column actually gets hidden